javascript/packages/orchestrator/src/networkNode.ts

```diff
@@ -276,7 +276,7 @@ export class NetworkNode implements NetworkNodeInterface {
       ]);
       if (resp instanceof Error) {
         // use `undefined` metrics values in `equal` comparisons as `0`
-        if (timedout && comparator === "equal" && desiredMetricValue === 0)
+        if (timedout && comparator === "equal" && desiredMetricValue === 0 && value === undefined)
```

Comment:
Yes, the edge case is **only** valid when the desired value is `0` because prometheus doesn't reports some metrics until they are incremented to `1` , that was the root of this workaround.

Reply:
Ah ok...